test/unit/plugin/v2/loader_test.rb

```diff
@@ -0,0 +1,196 @@
+  def test_activation
+    # Setup
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'meaning_by_path')
+    registry = Inspec::Plugin::V2::Registry.instance
+    loader = Inspec::Plugin::V2::Loader.new(omit_bundles: true)
+    loader.load_all
+    status = registry[:'inspec-meaning-of-life']
+
+    # Management methods for activation
+    assert_respond_to status, :activators, 'A plugin status should respond to `activators`'
+    assert_respond_to registry, :find_activators, 'Registry should respond to `find_activators`'
+    assert_respond_to loader, :activate, 'Loader should respond to `activate`'
+
+    # Finding an Activator
+    assert_kind_of Array, status.activators, 'status should have an array for activators'
+    assert_kind_of Array, registry.find_activators(), 'find_activators should return an array'
+    assert_equal 'Inspec::Plugin::V2::Activator', registry.find_activators()[0].class.name, 'find_activators should return an array of Activators'
+    activator = registry.find_activators(plugin_type: :mock_plugin_type, name: :'meaning-of-life-the-universe-and-everything')[0]
+    refute_nil activator, 'find_activators should find the test activator'
+    [ :plugin_name, :plugin_type, :activator_name, :activated, :exception, :activation_proc, :implementation_class ].each do |method_name|
+      assert_respond_to activator, method_name
+    end
+
+    # Activation preconditions
+    refute activator.activated, 'Test activator should start out unactivated'
+    assert_nil activator.exception, 'Test activator should have no exception prior to activation'
+    assert_nil activator.implementation_class, 'Test activator should not know implementation class prior to activation'
+    refute InspecPlugins::MeaningOfLife.const_defined?(:MockPlugin), 'impl_class should not be defined prior to activation'
+
+    loader.activate(:mock_plugin_type, 'meaning-of-life-the-universe-and-everything')
+
+    # Activation postconditions
+    assert activator.activated, 'Test activator should be activated after activate'
+    assert_nil activator.exception, 'Test activator should have no exception after activation'
+
+    # facts about the implementation class
+    impl_class = activator.implementation_class
+    refute_nil impl_class, 'Activation should set the implementation class'
+    assert_kind_of Class, impl_class, 'Should have a Class in the implementation class slot'
+    assert_includes impl_class.ancestors, Inspec::Plugin::V2::PluginBase, 'impl_class should derive from PluginBase'
+    assert_includes impl_class.ancestors, Inspec::Plugin::V2::PluginType::Mock, 'impl_class should derive from PluginType::Mock'
+    assert InspecPlugins::MeaningOfLife.const_defined?(:MockPlugin), 'impl_class should now be defined'
+
+  end
+
+  # load core plugins (or assert that we can interrogate the v1 registr(ies))
+  # output error but not raise when bad entry point
+
+  # it should handle a gem conflict correctly
```

Comment:
Should we remove these bits? Add them but use skip?

Reply:
These have become irrelevant, I've removed them - thanks